omod/src/main/java/org/openmrs/module/webservices/rest/Activator.java

```diff
@@ -28,6 +29,9 @@
 	@Override
 	public void started() {
 		log.info("Started the REST Web Service module");
+		
+		// Cache Swagger Spec
+		new SwaggerSpecificationCreator("host/openmrs/ws/rest").BuildJSON();
```

Comment:
If we could get the `host` and the `context path` in here, we could simplify this PR a little more.

Reply:
Did you intentionally hard code the above?

Reply:
Yes. I don't think we could get the actual host and the context path there. Let me know if we could. So therefore we cache it with some hard coded string as the base uri. Thereafter whenever some one call new `SwaggerSpecificationCreator("updated_string").BuildJSON()` (as done in [here](https://github.com/gayanW/openmrs-module-webservices.rest/blob/a72ba43ecf7392dd22747671a5071f47d02f56ae/omod/src/main/java/org/openmrs/module/webservices/rest/web/controller/SwaggerSpecificationController.java#L31-L65)) it will return the cached but with the `updated_string`.

Reply:
Did you try "WebConstants.WEBAPP_NAME" to get the context path?

Reply:
We need the name of the running host (domain).

Reply:
Since this is running from the server itself, can't you just use localhost?

Reply:
No then the client (try out) request will be sent to his own machines localhost. May we wait until we come to a conclusion on https://talk.openmrs.org/t/cant-make-requests-from-the-api-documentation/13145/11?u=gayanw

Then we could omit the `host/..` string altogether.
@psbrandt What about removing the host param from the spec?